### PR TITLE
Updating components to use v1beta3

### PR DESCRIPTION
**Pull request by nikhiljindal**
For https://github.com/GoogleCloudPlatform/kubernetes/issues/5475
- Setting the default version in clientcmds to v1beta3
- Updating api.Latest to point to v1beta3 instead of v1beta1

### CODE REVIEW

**Comment by nikhiljindal** [comment]
@bgrant0607 Verified that unit and integration tests pass.
Working on e2e tests.

File: pkg/client/request.go [hunk 1]
```diff
@@ -558,6 +558,7 @@ func (r *Request) DoRaw() ([]byte, error) {
 
 		var err error
 		r.req, err = http.NewRequest(r.verb, r.finalURL(), r.body)
+		glog.Errorf("REQUEST TO URL: %s", r.finalURL())
```
Comment (bgrant0607) [inline pkg/client/request.go]: Was this for debugging?

File: pkg/client/clientcmd/overrides.go [hunk 2]
```diff
@@ -141,7 +142,7 @@ func BindAuthInfoFlags(authInfo *clientcmdapi.AuthInfo, flags *pflag.FlagSet, fl
 // BindClusterFlags is a convenience method to bind the specified flags to their associated variables
 func BindClusterFlags(clusterInfo *clientcmdapi.Cluster, flags *pflag.FlagSet, flagNames ClusterOverrideFlags) {
 	flags.StringVarP(&clusterInfo.Server, flagNames.APIServer, flagNames.APIServerShort, "", "The address and port of the Kubernetes API server")
-	flags.StringVar(&clusterInfo.APIVersion, flagNames.APIVersion, "", "The API version to use when talking to the server")
+	flags.StringVar(&clusterInfo.APIVersion, flagNames.APIVersion, latest.Version, "The API version to use when talking to the server")
```
Comment (bgrant0607) [inline pkg/client/clientcmd/overrides.go]: How was the default version set to v1beta1 previously?

**Comment by bgrant0607** [comment]
What does this affect? Does it change the storage version?

File: pkg/client/clientcmd/overrides.go (hunk 2 above)
Comment (nikhiljindal) [inline pkg/client/clientcmd/overrides.go]: Its picking up latest.Version. I changed this just to be explicit.
Let me know if you want me to revert.

Comment (bgrant0607) [inline pkg/client/clientcmd/overrides.go]: This means you're leaving dead defaulting code all over. If we were to require the version to be specified via the flag (including flag default value) and/or in API objects (for create/update), then we should just require that and eliminate other defaults.

For example:
https://github.com/GoogleCloudPlatform/kubernetes/blob/b6f6de150afe566da8305e224d776d6dc7a6ce6e/pkg/kubectl/cmd/cmd.go#L344

Unfortunately, the means by which the version is set, from latest, mapping.APIVersion, flag, and/or config is confusing and spread all over.

You don't have to solve that problem now, though -- maybe revert and file an issue to come back to it.

**Comment by nikhiljindal** [comment]
Yes it affects storage version as well: https://github.com/GoogleCloudPlatform/kubernetes/blob/96a0a0d6189cf89f13f700bb5fa6c8f47aa109e5/pkg/master/master.go#L179

File: pkg/client/request.go (hunk 1 above)
Comment (bgrant0607) [inline pkg/client/request.go]: Please revert this change.

File: hack/e2e-test.sh [hunk 3]
```diff
@@ -15,6 +15,6 @@
 # limitations under the License.
 
 # Provided for backwards compatibility
-go run "$(dirname $0)/e2e.go" -v -build -up -test -down
+go run "$(dirname $0)/e2e.go" -v -up -test -down
```
Comment (bgrant0607) [inline hack/e2e-test.sh]: Why this change?

**Comment by bgrant0607** [comment]
Regenerate or revert the kubectl docs.

**Comment by nikhiljindal** [comment]
I think I have fixed all e2e tests now.

All but 3 tests are passing. The ones failing seem to be flaky on head as well:

```
Test: ReplicationController should serve a basic image on each replica with a private image
Error: Controller my-hostname-private-6e4c6f9f-da4e-11e4-998e-6c3be516adae: Gave up waiting for hostIP of replica 0 after 121.03349215 seconds

Test: ReplicationController should serve a basic image on each replica with a public image
Error: Controller my-hostname-basic-bbac7616-da4e-11e4-998e-6c3be516adae: Gave up waiting for hostIP of replica 0 after 121.041411761 seconds

Test: Monitoring verify monitoring pods and all cluster nodes are available on influxdb using heapster.
Error: monitoring using heapster and influxdb test failed
```

File: pkg/client/request.go (hunk 1 above)
Comment (nikhiljindal) [inline pkg/client/request.go]: Reverted

File: pkg/client/clientcmd/overrides.go (hunk 2 above)
Comment (nikhiljindal) [inline pkg/client/clientcmd/overrides.go]: Reverted the change

**Comment by nikhiljindal** [comment]
Cleaned up the PR and reverted the kubectl docs. Please have another look.

**Comment by vishh** [comment]
@nikhiljindal: The monitoring test has not been flaky for the last few days. I would suggest looking into why its failing.

File: cluster/vagrant/util.sh [hunk 4]
```diff
@@ -165,7 +165,7 @@ function verify-cluster {
     local count="0"
     until [[ "$count" == "1" ]]; do
       local minions
-      minions=$("${KUBE_ROOT}/cluster/kubectl.sh" get minions -o template -t '{{range.items}}{{.id}}:{{end}}')
+      minions=$("${KUBE_ROOT}/cluster/kubectl.sh" get minions -o template -t '{{range.items}}{{.id}}:{{end}}' --api-version=v1beta1)
```
Comment (bgrant0607) [inline cluster/vagrant/util.sh]: Is there a reason why you didn't change this similarly to validate-cluster.sh? The inability to test the change?

File: hack/e2e-internal/e2e-status.sh [hunk 5]
```diff
@@ -58,7 +58,7 @@ until [[ ${all_running} == 1 ]]; do
     echo "All pods never 'Running' in time." >&2
     exit 1
   fi
-  statuses=($(${KUBECTL} get pods --template='{{range.items}}{{.currentState.status}} {{end}}'))
+  statuses=($(${KUBECTL} get pods --template='{{range.items}}{{.currentState.status}} {{end}}' --api-version=v1beta1))
```
Comment (bgrant0607) [inline hack/e2e-internal/e2e-status.sh]: Was this hard to convert to v1beta3? .status.phase doesn't work? If non-trivial, we could leave to later, but I'd rather just clean these up now rather than leave a lot of v1beta1 uses in the code.

File: hack/e2e-suite/services.sh [hunk 6]
```diff
@@ -383,7 +384,7 @@ wait_for_pods "${svc3_name}" "${svc3_count}"
 svc3_pods=$(query_pods "${svc3_name}" "${svc3_count}")
 
 # Get the portal IP.
-svc3_ip=$(${KUBECTL} get services -o template '--template={{.portalIP}}' "${svc3_name}")
+svc3_ip=$(${KUBECTL} get services -o template '--template={{.portalIP}}' "${svc3_name}" --api-version=v1beta1)
```
Comment (bgrant0607) [inline hack/e2e-suite/services.sh]: v1beta3 would be .spec.portalIP

File: pkg/kubectl/cmd/get.go [hunk 7]
```diff
@@ -48,7 +48,7 @@ $ kubectl get replicationController web
 $ kubectl get -o json pod web-pod-13je7
 
 // Return only the status value of the specified pod.
-$ kubectl get -o template web-pod-13je7 --template={{.currentState.status}}
+$ kubectl get -o template web-pod-13je7 --template={{.currentState.status}} --api-version=v1beta1
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/get.go]: Please convert the example to v1beta3. .status.phase.

File: test/e2e/service.go [hunk 8]
```diff
@@ -166,7 +166,6 @@ var _ = Describe("Services", func() {
 	It("should provide RW and RO services", func() {
 		svc := api.ServiceList{}
 		err := c.Get().
-			Namespace(api.NamespaceDefault).
```
Comment (bgrant0607) [inline test/e2e/service.go]: What was the problem here?

File: cluster/vagrant/util.sh (hunk 4 above)
Comment (nikhiljindal) [inline cluster/vagrant/util.sh]: Yes. I decided to change them in a separate PR.
We will have to change these for deleting v1beta1

**Comment by nikhiljindal** [comment]
@vishh The monitoring test failed in the last build on jenkins with the same error: http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce/lastSuccessfulBuild/

File: test/e2e/service.go (hunk 8 above)
Comment (nikhiljindal) [inline test/e2e/service.go]: The generated path did not have "/namespaces/default" in it.

**Comment by bgrant0607** [comment]
LGTM.
